chaste_codegen/chaste_model.py

```diff
@@ -276,7 +279,12 @@ def _get_time_variable(self):
         self._set_is_metadata(time_variable, 'derived-quantity')
         try:
             # If the variable is in units that can be converted to millisecond, perform conversion
-            return self._model.convert_variable(time_variable, desired_units, DataDirectionFlow.INPUT)
+            time_var = self._model.convert_variable(time_variable, desired_units, DataDirectionFlow.INPUT)
+            self._model.rdf.add((time_var.rdf_identity, create_rdf_node((self._PYCMLMETA, 'derived-quantity')),
+                                create_rdf_node('yes')))
+            self._model.rdf.add((time_var.rdf_identity, create_rdf_node((self._PYCMLMETA, 'derived-quantity')),
+                                create_rdf_node('yes')))
+            return time_var
         except DimensionalityError:
```

Comment:
Rather than asserting I'd `raise ValueError(warning)`

Reply:
Fair point, this has been there for ages but raise is a better fit here.